game_engine.hpp

```diff
@@ -3,16 +3,16 @@
 
 #include <cstddef>
 #include <optional>
-#include <variant>
 #include <tuple>
+#include <variant>
 
```

Comment:
That reminds me to remove "<tuple>" that I added for BoardExtended that is now a struct.